shell-ui/src/toast/useMutationsHandler.ts

```diff
@@ -0,0 +1,108 @@
+import { ReactNode, useCallback, useEffect } from 'react';
+import { UseMutationResult } from 'react-query';
+import { useToast } from './useToast';
+
+export type MainMutationConfig<Data, Variables> = {
+  mutation: UseMutationResult<Data, unknown, Variables, unknown>;
+  name: string;
+  isPrimary: boolean;
+};
+
+type DependantMutationConfig<Data, Variables> = Omit<
+  MainMutationConfig<Data, Variables>,
+  'isPrimary'
+>;
+
+type DescriptionBuilder<Data> = {
+  data?: Data;
+  error?: unknown;
+  name: string;
+};
+
+type MutationsHandlerProps<Data, Variables> = {
+  mainMutation: MainMutationConfig<Data, Variables>;
+  dependantMutations?: DependantMutationConfig<Data, Variables>[];
+  messageDescriptionBuilder: (
+    successMutations: DescriptionBuilder<Data>[],
+    errorMutations: DescriptionBuilder<Data>[],
+  ) => ReactNode;
+  toastStyles?: React.CSSProperties;
+  onPrimarySuccess?: () => void;
```

Comment:
maybe we should rename this to mainMutation too ;) 
```suggestion
  onMainMutationSuccess?: () => void;
```